src/esp/gfx/GenericShader.cpp

```diff
@@ -122,12 +122,14 @@ enum { TextureLayer = 0 };
 }
 
 GenericShader::GenericShader(const Flags flags) : flags_(flags) {
-  MAGNUM_ASSERT_GL_VERSION_SUPPORTED(Magnum::GL::Version::GL410);
-
-  Magnum::GL::Shader vert{Magnum::GL::Version::GL410,
-                          Magnum::GL::Shader::Type::Vertex};
-  Magnum::GL::Shader frag{Magnum::GL::Version::GL410,
-                          Magnum::GL::Shader::Type::Fragment};
+  // MAGNUM_ASSERT_GL_VERSION_SUPPORTED(Magnum::GL::Version::GL410);
+#ifdef BUILD_WEBGL
+  Magnum::GL::Version glVersion = Magnum::GL::Version::GLES300;
+#else
+  Magnum::GL::Version glVersion = Magnum::GL::Version::GL410;
+#endif
```

Comment:
There's the same handy `CORRADE_TARGET_EMSCRIPTEN` macro that you can use for preprocessor branching -- no need to propagate `BUILD_WEBGL` from CMake either :) In this case, however, I think `MAGNUM_TARGET_WEBGL` would make more sense. [Docs](https://doc.magnum.graphics/magnum/Magnum_8h.html#define-members). These macros get defined whenever you include any Corrade or Magnum header.